src/SIL.Harmony.Core/IRemoteResourceService.cs

```diff
@@ -0,0 +1,27 @@
+﻿namespace SIL.Harmony.Core;
+
+/// <summary>
+/// interface to facilitate downloading of resources, typically implemented in application code
+/// the remote Id is opaque to the CRDT lib and could be a URL or some other identifier provided by the backend
+/// the local path returned for the application code to use as required, it could be a URL if needed also.
+/// </summary>
+public interface IRemoteResourceService
+{
+    /// <summary>
+    /// instructs application code to download a resource from the remote server
+    /// the service is responsible for downloading the resource and returning the local path
+    /// </summary>
+    /// <param name="remoteId">ID used to identify the remote resource, could be a URL</param>
+    /// <param name="localResourceCachePath">path defined by the CRDT config where the resource should be stored</param>
+    /// <returns>download result containing the path to the downloaded file, this is stored in the local db and not synced</returns>
+    Task<DownloadResult> DownloadResource(string remoteId, string localResourceCachePath);
```

Comment:
This feels better to me
```suggestion
    Task<DownloadResult> DownloadResource(RemoteResource resource, string localResourceCachePath);
```

Reply:
I actually wanted to avoid and keep the API symmetric (download using remoteId and get a local path, upload using a local path and get back a remoteId). When you upload a file that API returns you a `remoteId`, a handle if you will, which you can later use to download said file. If we pass in the whole object then that could break the assumptions as it should essentially always give you the same data for the same remoteId, but if you give it more then it can do something else.

Reply:
OK, yeah. I guess `RemoteResource` has a bunch of stuff on it and the remote ID is all that should be needed here 👍